pkg/controllers/disruption/consolidation.go

```diff
@@ -177,10 +181,7 @@ func (c *consolidation) computeConsolidation(ctx context.Context, candidates ...
 
 	if allExistingAreSpot &&
 		results.NewNodeClaims[0].Requirements.Get(v1beta1.CapacityTypeLabelKey).Has(v1beta1.CapacityTypeSpot) {
-		if len(candidates) == 1 {
-			c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, "Can't replace a spot node with a spot node")...)
-		}
-		return Command{}, nil
+		return c.computeSpotToSpotConsolidation(ctx, candidates, results, instanceTypesOptionsFilteredByPrice)
 	}
 
 	// We are consolidating a node from OD -> [OD,Spot] but have filtered the instance types by cost based on the
```

Comment:
If we're going to apply the 15 instance type minimum for cases where we are consolidating multiple spot nodes into a single spot node, then I question whether we should apply this restriction to cases where we have a mix of spot nodes and on-demand nodes since that's technically an even more available situation that we are consolidating from. 

Further, I'm realizing today that there's also no check ensuring that we have a minimum flexibility for cases where we consolidate from a single on-demand node to a spot node. That change would be a bit more tenuous since it's breaking existing behavior, but I'd like to get to a place where we can reason about everything in a consistent manner. 

Maybe it makes sense to say, "a consolidation to a spot node can only happen with minimum flexibility." 

cc: @njtran @tzneal

Reply:
Conclusion drawn from offline conversation: We are going to only implement minFlexibility as a concept on spot->spot consolidation for single node consolidation. Since we don’t currently have flexibility defined for od->spot multi-node or mixed spot/od->spot multi-node, it doesn’t feel conceptually reasonable to only require minFlexibility in the spot->spot multi-node edge case.

And the latest code is in line with this approach.